src/Core/Grid/Query/CurrencyQueryBuilder.php

```diff
@@ -118,11 +126,19 @@ private function getQueryBuilder(array $filters)
                 'cs',
                 'c.`id_currency` = cs.`id_currency`'
             )
+            ->innerJoin(
+                'c',
+                $this->dbPrefix . 'currency_lang',
+                'cl',
+                'c.`id_currency` = cl.`id_currency`'
+            )
         ;
         $qb->andWhere('cs.`id_shop` IN (:shops)');
+        $qb->andWhere('cl.`id_lang` = (:lang)');
         $qb->andWhere('c.`deleted` = 0');
 
         $qb->setParameter('shops', $this->contextShopIds, Connection::PARAM_INT_ARRAY);
+        $qb->setParameter('lang', $this->contextLangId, PDO::PARAM_INT);
```

Comment:
what's the use of this third param? avoiding a cast to string?

Reply:
If we can provide the type of data sent to the request, we should send it as parameter. Validation rules would be triggered to confirm we don't send weird stuff to the request.

Reply:
Ok nice!